core/src/test/java/io/grpc/internal/ServiceConfigErrorHandlingTest.java

```diff
@@ -541,7 +571,17 @@ public NameResolver newNameResolver(final URI targetUri, NameResolver.Args args)
         return null;
       }
       assertEquals(DEFAULT_PORT, args.getDefaultPort());
-      FakeNameResolver resolver = new FakeNameResolver(args.getServiceConfigParser());
+      FakeNameResolver resolver;
+      if (enableServiceConfigParsing) {
```

Comment:
An "old" NameResolver that doesn't do service config parsing should call `listener.onAddresses()`, not `onResult()`. This makes sure the compatibility logic in `Listener2#onAddresses()` is exercised.